### PR TITLE
RHDEVDOCS-6275: Adding missing properties in Argo CD CR and Repo server properties tables for GitOps 1.15

### PR DESCRIPTION
**Pull request by Dhruv-Soni11**
Version(s):

GitOps 1.15

Issue:

https://issues.redhat.com/browse/RHDEVDOCS-6275

Link to docs preview:

[Argo CD custom resource properties](https://85114--ocpdocs-pr.netlify.app/openshift-gitops/latest/argocd_instance/argo-cd-cr-component-properties.html#argo-cd-properties_argo-cd-cr-component-properties)

[Repo server properties](https://85114--ocpdocs-pr.netlify.app/openshift-gitops/latest/argocd_instance/argo-cd-cr-component-properties.html#argo-repo-server-properties_argo-cd-cr-component-properties)

QE review:
- [x] QE has approved this change.

SME review: @svghadi 
QE review: @varshab1210 
Peer review: @skrthomas 

Additional information:

### CODE REVIEW

File: modules/gitops-argo-cd-properties.adoc [hunk 1]
```diff
@@ -51,6 +56,12 @@ a|* `<processors.operation>` - The number of operation processors.
   * `<env>` - Environment to set for the application controller workloads.
   * `<sourceNamespaces>` - The list of non-control plane namespaces for creating and managing Argo CD `Application` resources in target namespaces.
   * `<extraCommandArgs>` - List of arguments added to the existing arguments set by the Operator.
+  * `annotations` - The number of custom annotations that are added to pods deployed by the Operator.
+  * `labels` - The number of custom labels that are added to pods deployed by the Operator.
```
Comment (svghadi) [inline modules/gitops-argo-cd-properties.adoc:60]:
```suggestion
  * `annotations` - List of custom annotations to add to pods deployed by the Operator. This field is optional.
  * `labels` - List of custom labels to add to pods deployed by the Operator. This field is optional.
```
Can we update with this for all labels & annotations properties in this PR?

File: modules/gitops-argo-cd-properties.adoc [hunk 2]
```diff
@@ -51,6 +56,12 @@ a|* `<processors.operation>` - The number of operation processors.
...
+  * `volumes` - The number of addition volumes configured for the Argo CD Application Controller component. This field is optional.
+  * `volumeMounts` - The number of addition volume mounts configured for the Argo CD Application Controller component. This field is optional.
```
Comment (svghadi) [inline modules/gitops-argo-cd-properties.adoc:62]:
```suggestion
  * `volumes` - List of addition volumes configured for the Argo CD Application Controller component. This field is optional.
  * `volumeMounts` - List of addition volume mounts configured for the Argo CD Application Controller component. This field is optional.
```
Can we update with this for all volumes & volumeMounts properties in this PR?

File: modules/gitops-argo-cd-properties.adoc [hunk 3]
```diff
@@ -158,6 +169,14 @@ a|* `<autoscale>` - Server autoscale configuration options.
...
   * `<env>` - Environment to set for the server workloads.
   * `<enabled>` - The flag to enable Argo CD server during the Argo CD installation.
+  * `volumes` - The number of addition volumes configured for the Argo CD Application Controller component. This field is optional.
+  * `volumeMounts` - The number of addition volume mounts configured for the Argo CD Application Controller component. This field is optional.
+  * `initContainers` - List of `init` containers for the Argo CD Application Controller component. This field is optional.
+  * `sidecarContainers` - List of `sidecar` containers for the Argo CD Application Controller component. This field is optional.
+  * `annotations` - The number of custom annotations that are added to pods deployed by the Operator.
+  * `labels` - The number of custom labels that are added to pods deployed by the Operator.
+  * `enableRolloutsUI` - The extension used for enabling the Argo Rollouts UI in Argo CD when it is set to
+  true.
```
Comment (svghadi) [inline modules/gitops-argo-cd-properties.adoc:179]:
```suggestion
  * `enableRolloutsUI` - Enables the Argo Rollouts UI extension in Argo CD when it is set to true. Default is false.
```

**Review by svghadi — CHANGES_REQUESTED** [review]

File: modules/gitops-argo-cd-properties.adoc (hunk 1 above)
Comment (Dhruv-Soni11) [reply modules/gitops-argo-cd-properties.adoc:60]: Fixed.

File: modules/gitops-argo-cd-properties.adoc (hunk 2 above)
Comment (Dhruv-Soni11) [reply modules/gitops-argo-cd-properties.adoc:62]: Fixed.

File: modules/gitops-repo-server-properties.adoc [hunk 4]
```diff
@@ -30,4 +30,6 @@ The following properties are available for configuring the repo server component
 |`volumeMounts` | `__<empty>__` |Configures additional volume mounts used for the Repo server deployment. This field is optional.
 |`replicas` | `__<empty>__` |The number of replicas for the Argo CD Repo server. Must be greater than or equal to `0`.
 |`remote` | `__<empty>__` |Specifies the remote URL of the Repo server container.
+|`annotations` | `__<empty>__` |The number of custom annotations that are added to pods deployed by the Operator.
```
Comment (varshab1210) [inline modules/gitops-repo-server-properties.adoc:33]:
```suggestion
|`annotations` | `__<empty>__` |List of custom annotations that are added to pods deployed by the Operator.
```

File: modules/gitops-repo-server-properties.adoc [hunk 5]
```diff
@@ -30,4 +30,6 @@ The following properties are available for configuring the repo server component
...
+|`labels` | `__<empty>__` |The number of custom labels that are added to pods deployed by the Operator.
```
Comment (varshab1210) [inline modules/gitops-repo-server-properties.adoc:34]:
```suggestion
|`labels` | `__<empty>__` |List of custom labels that are added to pods deployed by the Operator.
```

Comment (Dhruv-Soni11) [reply modules/gitops-repo-server-properties.adoc]: Fixed.

File: modules/gitops-repo-server-properties.adoc (hunk 4 above)
Comment (Dhruv-Soni11) [reply modules/gitops-repo-server-properties.adoc]: Fixed.

**Review by svghadi — APPROVED** [review]
/lgtm

**Review by skrthomas — COMMENTED** [review]
/lgtm

**Comment by varshab1210** [comment]
LGTM

File: modules/gitops-argo-cd-properties.adoc [hunk 6]
```diff
@@ -29,6 +29,11 @@ a|* `<image>` - The container image for the ApplicationSet Controller. This prop
   * `scmProviders` - The URLs of the allowed Source Code Manager (SCM) providers.
   * `<scmRootCAConfigMap>` - The name of the config map that stores the Gitlab SCM Provider's TLS certificate that will be mounted on the Application Set Controller at the "/app/tls/scm/cert" path.
   * `<webhookServer>` - Defines the available options for the ApplicationSet webhook server.
+  * `annotations` - List of custom annotations to add to pods deployed by the Operator. This field is optional.
+  * `labels` - List of custom labels to add to pods deployed by the Operator. This field is optional.
+  * `volumes` - List of addition volumes configured for the Argo CD ApplicationSet Controller component. This field is optional.
```
Comment (dfitzmau) [inline modules/gitops-argo-cd-properties.adoc:34]:
```suggestion
  * `volumes` - List of addition volumes configured for the Argo CD `ApplicationSet` Controller component. This field is optional.
```

File: modules/gitops-argo-cd-properties.adoc [hunk 7]
```diff
@@ -29,6 +29,11 @@ a|* `<image>` - The container image for the ApplicationSet Controller. This prop
...
+  * `volumeMounts` - List of addition volume mounts configured for the Argo CD ApplicationSet Controller component. This field is optional.
```
Comment (dfitzmau) [inline modules/gitops-argo-cd-properties.adoc:35]:
```suggestion
  * `volumeMounts` - List of addition volume mounts configured for the Argo CD `ApplicationSet` Controller component. This field is optional.
```

File: modules/gitops-argo-cd-properties.adoc [hunk 8]
```diff
@@ -158,6 +169,13 @@ a|* `<autoscale>` - Server autoscale configuration options.
   * `<logFormat>` - The log format used by the Argo CD server component. Valid options are `text` and `json`.
   * `<env>` - Environment to set for the server workloads.
   * `<enabled>` - The flag to enable Argo CD server during the Argo CD installation.
+  * `volumes` - List of addition volumes configured for the Argo CD Application Controller component. This field is optional.
+  * `volumeMounts` - List of addition volume mounts configured for the Argo CD Application Controller component. This field is optional.
+  * `initContainers` - List of `init` containers for the Argo CD Application Controller component. This field is optional.
+  * `sidecarContainers` - List of `sidecar` containers for the Argo CD Application Controller component. This field is optional.
+  * `annotations` - List of custom annotations to add to pods deployed by the Operator. This field is optional.
+  * `labels` - List of custom labels to add to pods deployed by the Operator. This field is optional.
+  * `enableRolloutsUI` -   Enables the Argo Rollouts UI extension in Argo CD when it is set to true. The default value is set to false.
```
Comment (dfitzmau) [inline modules/gitops-argo-cd-properties.adoc:178]:
```suggestion
  * `enableRolloutsUI` -   When the parameter is set to `true`, the parameter enables the Argo Rollouts UI extension in Argo CD. The default value is set to `false`.
```

File: modules/gitops-argo-cd-properties.adoc (hunk 7 above)
Comment (Dhruv-Soni11) [reply modules/gitops-argo-cd-properties.adoc:35]: Fixed.

File: modules/gitops-argo-cd-properties.adoc (hunk 8 above)
Comment (Dhruv-Soni11) [reply modules/gitops-argo-cd-properties.adoc:178]: Fixed.

**Pull request merged**